addon/-private/system/model/internal-model.js

```diff
@@ -111,28 +123,28 @@ export default class InternalModel {
     this.store = store;
     this._data = data || new EmptyObject();
     this.modelName = type.modelName;
-    this.dataHasInitialized = false;
     this._loadingPromise = null;
     this._recordArrays = undefined;
-    this.currentState = RootState.empty;
-    this.isReloading = false;
+    this._isDestroying = false;
     this._isDestroyed = false;
     this.isError = false;
-    this.error = null;
+
+    this.resetRecord();
 
     // caches for lazy getters
     this.__deferredTriggers = null;
     this._references = null;
     this._recordReference = null;
-    this.__inFlightAttributes = null;
     this.__relationships = null;
-    this.__attributes = null;
     this.__implicitRelationships = null;
+
+    // Used for coloring during BFS
```

Comment:
should likely explain just a-bit more. BFS doesn't help us understand why/where this is used. Likely would be enough to say "during marking phase of unloading"